### DISCUSSION TITLE
Parameter without default value in a Modelica model

### DISCUSSION THREAD

**Discussion by pierre-haessig** [OpenModelica/OpenModelica#10385] [General]
I'm forwarding here the question I asked on StackOverflow: https://stackoverflow.com/q/75663745/2822346.

I thought my interrogation on "Parameter without default value in a Modelica model" was related to the Modelica language specification in general. However, the comments and the answer I got from Hans Olsson showed me it was also related to the implementation. In particular, it seems that OpenModelica is more "lenient" with parameters without values (it emits a warning and uses a instead start value=0.0) whereas Dymola fails.

I  think I prefer the stricter behavior of Dymola. But of course if many models rely on the warning...

**Comment by casella** [comment]
MLS 3.5 [Section 8.6](https://specification.modelica.org/maint/3.5/equations.html#initialization-initial-equation-and-initial-algorithm) is very clear:
> If a parameter has a _modifier_ for the start attribute, does not have `fixed = false`, and neither has a binding equation nor is part of a record having a binding equation, the modifier for the start attribute can be used to add a parameter binding equation assigning the parameter to that start value. In this case a diagnostic message is recommended in a simulation model.

So, according to MSL 3.5, this model should be rejected:
```modelica
model M
  parameter Real p;
end M;
```
MLS removed the default start attribute 0.0 from the predefined Real definition and accordingly [changed the definition in 8.6](https://specification.modelica.org/maint/3.6/equations.html#initialization-initial-equation-and-initial-algorithm):

> If a parameter has a _value_ for the start-attribute, does not have fixed = false, and neither has a binding equation nor is part of a record having a binding equation, the value for the start-attribute can be used to add a parameter binding equation assigning the parameter to that start value. In this case a diagnostic message is recommended in a simulation model.

but the net effect is the same.

I opened #10386 on this topic. #pierre-haessig thanks for reporting!

**Reply by pierre-haessig** [reply]
Thanks for the explanation and opening the ticket!
